<!DOCTYPE html>
<html>
<head><title>Atlantis</title> {% extends "bootstrap/base.html" %}{% block content %}
<link href="https://www.w3schools.com/w3css/4/w3.css" rel="stylesheet">
<link rel="stylesheet" href="/static/css/app_styles.css">
<script src="/static/js/app_script.js" type="text/javascript"></script>
</head>

<body>

  <nav class="navbar bg-primary navbar-dark" role="navigation" style='height: 45px; border-radius: 0px; background-color: #1463BD'>
      <div class="container-fluid">
        <div class="navbar-header">
          <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav">
                <li style='margin-right: 5px'><font face="garamond" size="6"> Atlantis: RNA-seq visualisation </font></li>
                <li style='border-left: solid #FFFFFF; color:#1463BD'><font face="garamond" size="6">_</font></li>
          </div>
        </div>
      </div>

        <p style='position:absolute; top:12px;right:20px; color:white'><b><u> Version:</u></b>  {{version}}</p>
    </nav>

            <p style='color:red; text-align:center; margin: 0 auto; width:500px; margin-top: 20px'>
            {% with messages = get_flashed_messages() %}
               {% if messages %}
                  {% for message in messages %}
                     {{ message }}
                  {% endfor %}
               {% endif %}
            {% endwith %}
            </p>


            <form action="" class="container" enctype="multipart/form-data" method="post">
                <!-- Based on the 'status' variable, different parts of the content below will show up. The content has been
                precoded to show up sequentially and the next piece of content will not show up if an error occured during the
                'POST' action-->

                <div style='text-align: center; margin: 0 auto; padding: 5px; border-radius: 10px; background-color: #1463BD; border-radius: 10px'
                class="d-block navbar-dark bg-primary text-white" id="content">
                <div style='text-align:center; margin: 0 auto; padding: 10px;border-radius: 10px; background-color: white'
                class="d-block navbar-dark bg-primary text-white" id="content">

                <h2 style='color:#1463BD'><font face="garamond"> Quality Check using Machine Learning </font> </h2>

                  <img src='/static/images/ml_plot.png' alt='Output of the Dimensionality Reduction Clustering'
                  style='border-radius: 10px; height: 450px; width: 550px'/>
                  <br><br>
                  <p style='color:grey'><font face="garamond">
                    By plotting the two dominant components of the gene distributions in each sample, we check the similarity of the samples. <br>
                    If a sample does not form a cluster with the rest of the samples, it may have a lab error or sequencing error.<br><br>

                    <b>Results from the PCA analysis:</b> <br>
                    1. Variance Ratio of the first component: {{first_var}} %<br>
                    2. Variance Ratio of the second component: {{second_var}} %<br>
                  </font> </p>

                <br><br>
                <hr style='background-color: #1463BD; height: 2px'>
                <h2 style='color:#1463BD'><font face="garamond"> Data Visualisations </font> </h2>

              <div style='height: 500px; width: 550px; text-align: center; margin: 0 auto; display:inline-block'>

                <img src='/static/images/scatter.png' alt='Gene Count Distributions'
                style='border-radius: 10px; height: 450px; width: 550px; display:inline-block'/>
                <br><br>
                <p style='color:grey'><font face="garamond">
                  By comparing the distributions and identifying areas where the genes have clustered,<br>
                  we can infer particular samples that have a significant change in its gene expression.
                </font> </p>
              </div>

              <div style='height: 500px; width: 550px; text-align: center; margin: 0 auto; display:inline-block'>

                <img src='/static/images/box_plot.png' alt='Box Plots'
                style='border-radius: 10px; height: 450px; width: 550px; display:inline-block'/>
                <br><br>
                <p style='color:grey'><font face="garamond">
                  The box plots also help us compare the general distribution of the gene counts.
                  Specifically, it helps us compare the means and the confidence intervals of the gene count distributions.

                </font> </p>
              </div> <br> <br>
                <hr style='text-align: center; margin: 0 auto; background-color: #1463BD; height: 2px; width: 80%'>
                <br>

                <div style='height: 500px; width: 550px; text-align: center; margin: 0 auto; display:inline-block'>

                  <img src='/static/images/median_plot.png' alt='Median Plots'
                  style='border-radius: 10px; height: 450px; width: 550px; display:inline-block'/>
                  <br><br>
                  <p style='color:grey'><font face="garamond">
                    By comparing the median gene counts for samples in different conditions, we can infer whether there
                    is a significant overexpression or underexpression in one of the conditions.

                  </font> </p>
                </div>

                <div style='height: 500px; width: 550px; text-align: center; margin: 0 auto; display:inline-block'>

                  <img src='/static/images/volcano.png' alt='Median Plots'
                  style='border-radius: 10px; height: 450px; width: 550px; display:inline-block'/>
                  <br><br>
                  <p style='color:grey'><font face="garamond">
                    By understanding the balance between the log-fold change and the p-values, we can infer valuable causal relationships
                    between the given condition and the average gene expression.

                  </font> </p>
                </div>

                <br><br>
                <hr style='text-align: center; margin: 0 auto; background-color: #1463BD; height: 2px; width: 100%'>
                <br> <h2 style='color:#1463BD'><font face="garamond"> Filtering and Mapping Outlier Genes </font> </h2>

                <p style='color:grey'><font face="garamond">
                  Number of Filtered Genes in the dataset with p-value < 0.05 and mod(log-fold change) > 2 are: {{filtered_num}}

                </font> </p>

                <div style='height: 500px; width: 550px; text-align: center; margin: 0 auto; display:inline-block'>

                  <img src='/static/images/gene_map.png' alt='Median Plots'
                  style='border-radius: 10px; height: 450px; width: 550px; display:inline-block'/>
                  <br><br>
                  <p style='color:grey'><font face="garamond">
                    By understanding the differences between the gene counts for the filtered genes, we can identify any faults or problems with the most outlying genes.

                  </font> </p>
                </div>

                <div style='height: 500px; width: 550px; text-align: center; margin: 0 auto; display:inline-block'>

                  <img src='/static/images/heatmap.png' alt='Median Plots'
                  style='border-radius: 10px; height: 450px; width: 550px; display:inline-block'/>
                  <br><br>
                  <p style='color:grey'><font face="garamond">
                    Visualising the gene counts as a heat map helps us understand the relative variation of the gene counts across all the samples.
                  </font> </p>
                </div>

                </div>

              <br>
              </form>
{% endblock %}
    </body>
</html>
